packages/eslint-plugin/docs/rules/strict-void-return.mdx

```diff
@@ -0,0 +1,496 @@
+---
+description: 'Disallow passing a value-returning function in a position accepting a void function.'
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/strict-void-return** for documentation.
+
+## Rule Details
+
+TypeScript considers functions returning a value to be assignable to a function returning void.
+Using this feature of TypeScript can lead to bugs or confusing code.
+
+## Examples
+
+### Return type unsafety
+
+Passing a value-returning function in a place expecting a void function can be unsound.
+
+<Tabs>
+<TabItem value="❌ Incorrect">
+
+```ts
+const getNothing: () => void = () => 2137;
+const getString: () => string = () => 'Hello';
+const getStringMaybe = Math.random() > 0.1 ? getNothing : getString;
+const maybeString = getStringMaybe();
```

Comment:
(praise) This version of the example is much easier to follow. Thanks! 🙂 

(nit) why not inline the `getStringMaybe` variable still?
```suggestion
const maybeString = Math.random() > 0.1 ? getNothing() : getString();
```